RxSwift/Traits/PrimitiveSequence+Zip+Collection.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  PrimitiveSequence+Zip+Collection.swift
+//  RxSwift-iOS
+//
+//  Created by yuzushioh on 2018/01/23.
+//  Copyright © 2018 Krunoslav Zaher. All rights reserved.
+//
+
+// MARK: zip + collection
+
+extension PrimitiveSequenceType where TraitType == SingleTrait {
+    /**
+     Merges the specified observable sequences into one observable sequence by using the selector function whenever all of the observable sequences have produced an element at a corresponding index.
+     
+     - parameter resultSelector: Function to invoke for each series of elements at corresponding indexes in the sources.
+     - returns: An observable sequence containing the result of combining elements of the sources using the specified result selector function.
+     */
+    public static func zip<C: Collection, R>(_ collection: C, _ resultSelector: @escaping ([ElementType]) throws -> R) -> PrimitiveSequence<TraitType, R> where C.Iterator.Element == PrimitiveSequence<TraitType, ElementType> {
```

Comment:
There is one additional condition that needs to be checked here.

```swift
if collection.isEmpty {
       return PrimitiveSequence<TraitType, R>(raw: .just(resultSelector([])))
}
```

And unit test added.